tests/validation.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { validate } from "../src/sgb-xi/validation";
+import { payload1, payload2, payload3 } from "./samples/billingPayloads";
+
+describe("validation", () => {
+
+    it("shows no errors or warnings for billing with invoice type 1", () => {
+        const result = validate(payload1.invoices, payload1.billingData);
+
+        expect(result.errors).toEqual([]);
+        expect(result.warnings).toEqual([]);
+    });
+
+    it("shows no errors or warnings for billing with invoice type 2", () => {
+        const result = validate(payload2.invoices, payload2.billingData);
+
+        expect(result.errors).toEqual([]);
+        expect(result.warnings).toEqual([{
+            code: "textIsTruncated",
+            params: {
+                key: "name",
+                maxLength: "30",
+            },
+            path: ["invoices", 0, "leistungserbringer"],
+            message: "The property \"name\" will be truncated, because it is longer than 30 characters.",
+        }]);
+    });
+
+    it("shows no errors or warnings for billing with invoice type 3", () => {
+        const result = validate(payload3.invoices, payload3.billingData);
+
+        expect(result.errors).toEqual([]);
+        expect(result.warnings).toEqual([]);
+    });
+
+    it("shows several errors for incorrect billingData", () => {
+        const result = validate(payload3.invoices, {
+            ...payload3.billingData,
+            //@ts-ignore
```

Comment:
what's this?

Reply:
It tells TSLint to ignore the type error on the next line, as I intentionally violate the rule that rechnungsart has to be defined.